doc/man1/openssl.pod

```diff
@@ -1300,13 +1302,37 @@ respectively.
 
 =item B<-engine> I<id>
 
-Use the engine identified by I<id> and use all the methods it
-implements (algorithms, key storage, etc.), unless specified otherwise in
-the command-specific documentation or it is configured to do so, as described
-in L<config(5)/Engine Configuration Module>.
+Load the engine identified by I<id> and use all the methods it implements
+(algorithms, key storage, etc.), unless specified otherwise in the
+command-specific documentation or it is configured to do so, as described in
+L<config(5)/Engine Configuration>.
+
+The engine will be used for key ids specified with B<-key> and similar
+options when an option like B<-keyform engine> is given.
 
 =back
 
+Options specifying keys, like B<-key> and similar, can use the generic
+OpenSSL engine key loading URI scheme C<org.openssl.engine:> to retrieve
+private keys and public keys.  The URI syntax is as follows, in simplified
+form:
+
+    org.openssl.engine:{engineid}:{keyid}
+
+Where C<{engineid}> is the identity/name of the engine, and C<{keyid}> is a
+key identifier that's acceptable by that engine.  For example, when using an
+engine that interfaces against a PKCS#11 implementation, the generic key URI
+would be something like this:
+
+    -key org.openssl.engine:pkcs11:object=some-private-key;pin-value=1234
+
```

Comment:
Here we have the unfortunately confusing situation that `pkcs11:` needs to be given twice!
Once as the engineid, and then as RFC 7512 PKCS#11 scheme:
```
-key org.openssl.engine:pkcs11:pkcs11:object=some-private-key;pin-value=1234
```

libp11 also allows for simple keyids such as `id_cafe01234`, so the key arg becomes
```
-key org.openssl.engine:pkcs11:id_cafe01234
```

Reply:
With that particular example, it seems that `org.openssl.engine:pkcs11:label_some-private-key` is the corresponding simple key id...  that's at least how I'm reading the pkcs11.so source...

Reply:
EDited